src/main/java/uk/gov/justice/hmiprobation/casesampler/services/CaseListService.kt

```diff
@@ -1,37 +1,23 @@
 package uk.gov.justice.hmiprobation.casesampler.services
 
-import uk.gov.justice.hmiprobation.casesampler.dto.Case
+import uk.gov.justice.hmiprobation.casesampler.dto.PrimaryCaseSample
+import uk.gov.justice.hmiprobation.casesampler.dto.PrimaryCaseSampleProvisional
 import uk.gov.justice.hmiprobation.casesampler.utils.calculateSampleSize
 
-class CaseListService() {
 
-    /**
-     * @param longlist a primary case sample
-     * @return shortlist (a Primary Case Sample Provisional)
-     */
-    fun process(sampleSize: Int, longlist: List<Case>) {
+class CaseListService(val samplePicker: SamplePicker) {
 
-        val groupedByStratification = longlist.asSequence()
-                .filter { it.isExcluded() }
+    fun process(sampleSize: Int, bufferPercentage: Double, longlist: PrimaryCaseSample): PrimaryCaseSampleProvisional {
+
+        val casesByStratum = longlist.asSequence()
```

Comment:
Just wondering why a sequence here... the sequence cannot short circuit due to the groupBy, are there other benefits?

Reply:
There are 2 filters though? I suspect InteliJ would suggest it